components/HospitalCard.js

```diff
@@ -28,6 +28,17 @@ export default function HospitalCard(props) {
   const bgNotAvailable = useColorModeValue('red.100', 'red.700')
   const bgDistanceAvailable = useColorModeValue('gray.200', 'gray.600')
   const bgDistanceNotAvailable = useColorModeValue('red.200', 'red.600')
+
+  const locationAction = {
+    ...(onLocationClick
+      ? {
+          onClick: onLocationClick,
+        }
+      : {
+          href: generateGoogleMapsLink(hospital.name),
+        }),
+  }
```

Comment:
Hmmmm. I think this is better for us to do the conditional on the button component instead. Because when the button act like an `<a>` tag, they need some other props like `target`, `rel` & `href`. Maybe write something like this:

```js
{typeof onLocationClick === 'function' ? (
  <Button
    w="full"
    mb={[2, 0]}
    mr={{ base: '2' }}
    size="sm"
    onClick={onLocationClick}
  >
    <span style={{ marginRight: 5 }} aria-label="peta">
      📍
    </span>{' '}
    Lihat Lokasi
  </Button>
) : (
  <Button
    w="full"
    as="a"
    mb={[2, 0]}
    mr={{ base: '2' }}
    target="_blank"
    rel="noreferrer"
    size="sm"
    href={generateGoogleMapsLink(hospital.name)}
  >
    <span style={{ marginRight: 5 }} aria-label="peta">
      📍
    </span>{' '}
    Lihat Lokasi
  </Button>
)}
```

But, please test it on your local first. 🙇🏻

Reply:
Woops, sorry, didn't realize that's an anchor. Will update it  🙏